src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -62,10 +65,25 @@ bool isValidElementTypeAndRank(Value val, bool donotCheckRank) {
       if (donotCheckRank)
         return true;
       // Rank must be in range of (0, 4].
-      if (!valueType.hasRank())
+      if (!valueType.hasRank()) {
+        std::string message = "Value does not have rank.";
+        emitLegalityCheckMessage(op, StringRef(message));
         return false;
+      }
       int64_t rank = valueType.getRank();
-      return ((rank > 0) && (rank <= 4));
+      if ((rank > 0) && (rank <= 4)) {
+        return true;
+      } else {
+        std::string message =
+            "Rank " + std::to_string(rank) +
+            " is not supported. zAIU only supports rank in range of (0, 4].";
+        emitLegalityCheckMessage(op, StringRef(message));
+        return false;
+      }
+    } else {
+      std::string message = "Element type is not F16 or F32.";
+      emitLegalityCheckMessage(op, StringRef(message));
+      return false;
```

Comment:
I recommend to change `emitLegalityCheckMessage` to return `false`, e.g `bool emitLegalityCheckMessage(Operation *op, StringRef message)`, and then we can write:
```
return emitLegalityCheckMessage(op, StringRef(message));
```

Reply:
Done. Thanks. Also changed the function name.